contracts/verifiers/UniversalVerifier.sol

```diff
@@ -1,128 +1,51 @@
-
-    /// @dev Main storage structure for the contract
-    struct UniversalVerifierStorage {
-        // user -> ( requestID -> proof )
-        mapping(address => mapping(uint64 => Proof)) proofs;
-        mapping(uint64 => IUniversalVerifier.ZKPRequest) requests;
-        uint64[] requestIds;
-        mapping(address => uint64[]) userRequestIds;
-        mapping(ICircuitValidator => bool) whitelistedValidators;
-    }
-
-    // keccak256(abi.encode(uint256(keccak256("iden3.storage.UniversalVerifier")) - 1)) & ~bytes32(uint256(0xff));
-    bytes32 private constant UniversalVerifierStorageLocation =
-        0x0c87ac878172a541d6ba539a4e02bbe44e1f3a504bea30ed92c32fb1517db700;
-
-    /// @dev Get the main storage using assembly to ensure specific storage location
-    function _getUniversalVerifierStorage()
-        private
-        pure
-        returns (UniversalVerifierStorage storage $)
-    {
-        assembly {
-            $.slot := UniversalVerifierStorageLocation
-        }
-    }
-
-    uint256 constant REQUESTS_RETURN_LIMIT = 1000;
-
+contract UniversalVerifier is
+    Ownable2StepUpgradeable,
+    RequestOwnership,
+    RequestDisable,
+    ValidatorWhitelist
+{
     /**
      * @dev Version of contract
      */
     string public constant VERSION = "1.0.1";
 
-    /// @dev Key to retrieve the linkID from the proof storage
-    string constant LINKED_PROOF_KEY = "linkID";
-
     /// @dev Event emitted upon submitting a ZKP request
     event ZKPResponseSubmitted(uint64 indexed requestId, address indexed caller);
 
     /// @dev Event emitted upon adding a ZKP request
     event ZKPRequestSet(
         uint64 indexed requestId,
-        address indexed controller,
+        address indexed requestOwner,
         string metadata,
+        address validator,
         bytes data
     );
 
-    /// @dev Linked proof custom error
-    error LinkedProofError(
-        string message,
-        uint64 requestId,
-        uint256 linkID,
-        uint64 requestIdToCompare,
-        uint256 linkIdToCompare
-    );
-
-    /// @dev Modifier to check if the caller is the owner or controller of the ZKP request
-    modifier onlyOwnerOrController(uint64 requestId) {
-        require(
-            msg.sender == _getUniversalVerifierStorage().requests[requestId].controller ||
-                msg.sender == owner(),
-            "Only owner or controller can call this function"
-        );
-        _;
-    }
-
-    /// @dev Modifier to check if the ZKP request is enabled
-    modifier requestEnabled(uint64 requestId) {
-        require(
-            !_getUniversalVerifierStorage().requests[requestId].isDisabled,
-            "Request is disabled"
-        );
-        _;
-    }
-
-    /// @dev Modifier to check if the validator is whitelisted
-    modifier isWhitelistedValidator(ICircuitValidator validator) {
+    /// @dev Modifier to check if the caller is the contract Owner or ZKP Request Owner
+    modifier onlyRequestOwner(uint64 requestId) override {
```

Comment:
Should be something like `onlyOwnerOrRequestOwner`, because we already have `onlyRequestOwner` in RequestOwnership.sol, and this one checks also for Contract Owner.